src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -402,23 +402,29 @@ class GrowableArrayWithAllocator : public GrowableArrayView<E> {
 
   void push(const E& elem) { append(elem); }
 
-  E at_grow(int i, const E& fill = E()) {
+  template<typename... Args >
+  E& at_grow(int i, const Args&... args) {
     assert(0 <= i, "negative index %d", i);
     if (i >= this->_len) {
       if (i >= this->_capacity) grow(i);
-      for (int j = this->_len; j <= i; j++)
-        this->_data[j] = fill;
-      this->_len = i+1;
+      for (int j = this->_len; j <= i; j++) {
+        this->_data[j].~E();
+        ::new (&this->_data[j]) E(args...);
+      }
+      this->_len = i + 1;
     }
     return this->_data[i];
   }
 
-  void at_put_grow(int i, const E& elem, const E& fill = E()) {
+  template<typename... Args>
+  void at_put_grow(int i, const E& elem, const Args&... args) {
```

Comment:
Sorry, I'm not sure what you're missing here. This change enables in-place construction of the objects and the `Args...` doesn't have to be of type `E`. Is the issue with the default being removed?

Reply:
My point is that the old code was fine.  Copy-assigning the fill argument
(either defaulted or passed in) had to work (E can't be non-copyable), because
at_put_grow requires E to be copy-assignable in order to install the new value
at index i.

Reply:
>at_put_grow requires E to be copy-assignable in order to install the new value
at index i.

With the new code it doesn't seem to me that `at_put_grow` requires this, correct? I'd like to get rid of this requirement.

Reply:
See line#430: `this->_data[i] = elem;`.  That requires E to be copy-assignable.